classes/notification/PKPNotificationOperationManager.inc.php

```diff
@@ -385,19 +386,87 @@ private function sendNotificationEmail($request, $notification) {
 				$mail->setReplyTo($site->getLocalizedContactEmail(), $site->getLocalizedContactName());
 			}
 
+			$unsubscribeUrl = $this->getUnsubscribeNotificationUrl($request, $notification);
+
 			$mail->assignParams(array(
 				'notificationContents' => $this->getNotificationContents($request, $notification),
 				'url' => $this->getNotificationUrl($request, $notification),
-				'siteTitle' => $context?$context->getLocalizedName():$site->getLocalizedTitle()
+				'siteTitle' => $context?$context->getLocalizedName():$site->getLocalizedTitle(),
+				'notificationUnsubscribeUrl' => $unsubscribeUrl,
```

Comment:
Does this add it to every mail that is sent out? I think we only support unsubscribing from certain system emails and only those should have an unsubscribe link.(Others, like a discussion, can't be unsubscribed.)

Reply:
@NateWr it is added to every NOTIFICATION email send by the `sendNotificationEmail` function, which is filtered by the `getUserBlockedEmailedNotifications($userId, $contextId);`, which contain the information related to the unsubscribe action. 

The unsubscribe action it self, refers to all (and only those) `notification types` that are managed from the User -> Profile -> Notifications page. 

At the review fixes I have added all notification types that can be unsubscribed from, and the `$unsubscribeUrl` is assigned only if the notification is in that array